src/components/FormBuilder/FormBuilder.tsx

```diff
@@ -6,13 +6,12 @@ const FormBuilder: FormBuilderType<any> = (
   { fields, fieldOptions = {}, values, validate, handleSubmit, submitSection }) => {
 
   const setFormObject = (currentValues: any = {}) => {
-    const defaultValues = Object.keys(fieldOptions).reduce((defaultsObject, key) => {
-      const current = fieldOptions[key] || { defaultValue: null };
-
-      return { ...defaultsObject, [key]: current.defaultValue };
+    const defaults = fields.reduce((acc, field) => {
+      const options = fieldOptions[field] as FieldOptionsValueType<any>;
```

Comment:
now that you're using `fieldOptions[field]`, you can simplify it in the way we tried earlier:

```suggestion
      const { defaultValue = null } = fieldOptions[field] as FieldOptionsValueType<any>;
```
and remove the following line